### PR TITLE
Conv2d SPMD Support

### PR DESCRIPTION
**Pull request by lxd-cumt**
### PR Category
Distributed Strategy

### PR Types
New Features

### Description
Support conv2d spmd rules, add unitest

Pcard-66975

### CODE REVIEW

File: paddle/phi/infermeta/spmd_rules/conv2d.h [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+                            const DistMetaTensor& filter,
+                            const DistMetaTensor& output,
+                            const std::vector<int>& strides,
+                            const std::vector<int>& paddings,
+                            const std::string& padding_algorithm,
+                            const std::vector<int>& dilations,
+                            int groups,
+                            const std::string& data_format);
+
+SpmdInfo Conv2dInferSpmd(const DistMetaTensor& input,
```
Comment (jeff41404) [inline paddle/phi/infermeta/spmd_rules/conv2d.h]: should be `Conv2dGradInferSpmd` ?

File: paddle/phi/infermeta/spmd_rules/conv2d.cc [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+                            const DistMetaTensor& output,
+                            const std::vector<int>& strides,
+                            const std::vector<int>& paddings,
+                            const std::string& padding_algorithm,
+                            const std::vector<int>& dilations,
+                            int groups,
+                            const std::string& data_format) {
+        return Conv2dInferSpmdReverseBase(input, filter, output);
+    }
+
```
Comment (jeff41404) [inline paddle/phi/infermeta/spmd_rules/conv2d.cc:340]: shall we support `Conv2dGradInferSpmdBase` and `Conv2dGradInferSpmd`, or auto-parallel in dygraph will not support

File: paddle/phi/infermeta/spmd_rules/conv2d.cc [hunk 3]
```diff
@@ -0,0 +1,203 @@
+/* Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
```
Comment (From00) [inline paddle/phi/infermeta/spmd_rules/conv2d.cc]: 2023 -> 2024

**Review by From00 — COMMENTED** [review]
反向算子也需要切分推导

**Review by gongshaotian — APPROVED** [review]
LGTM for paddle::dialect::InferSymbolicShapeInterface

**Review by From00 — APPROVED** [review]

**Pull request merged**